applications/CompressiblePotentialFlowApplication/python_scripts/potential_flow_adjoint_solver.py

```diff
@@ -0,0 +1,90 @@
+        self.main_model_part.AddNodalSolutionStepVariable(KCPFApp.ADJOINT_AUXILIARY_VELOCITY_POTENTIAL)
+        self.main_model_part.AddNodalSolutionStepVariable(KratosMultiphysics.SHAPE_SENSITIVITY)
+
+        self.print_on_rank_zero("::[PotentialFlowAdjointSolver]:: ", "Variables ADDED")
+
+    def AddDofs(self):
+        KratosMultiphysics.VariableUtils().AddDof(KCPFApp.ADJOINT_VELOCITY_POTENTIAL, self.main_model_part)
+        KratosMultiphysics.VariableUtils().AddDof(KCPFApp.ADJOINT_AUXILIARY_VELOCITY_POTENTIAL, self.main_model_part)
+
+    def Initialize(self):
+        """Perform initialization after adding nodal variables and dofs to the main model part. """
+        if self.response_function_settings["response_type"].GetString() == "adjoint_lift_jump_coordinates":
+            self.response_function = KCPFApp.AdjointLiftJumpCoordinatesResponseFunction(self.main_model_part, self.response_function_settings)
+        else:
+            raise Exception("invalid response_type: " + self.response_function_settings["response_type"].GetString())
+
+        self.sensitivity_builder=KratosMultiphysics.SensitivityBuilder(self.sensitivity_settings,self.main_model_part, self.response_function)
+        self.sensitivity_builder.Initialize()
+
+        scheme = KratosMultiphysics.ResidualBasedAdjointStaticScheme(self.response_function)
+
+        builder_and_solver = KratosMultiphysics.ResidualBasedBlockBuilderAndSolver(self.linear_solver)
+        self.solver = KratosMultiphysics.ResidualBasedLinearStrategy(
+            self.main_model_part,
+            scheme,
+            self.linear_solver,
+            builder_and_solver,
+            self.settings["compute_reactions"].GetBool(),
+            self.settings["reform_dofs_at_each_step"].GetBool(),
+            self.settings["calculate_solution_norm"].GetBool(),
+            self.settings["move_mesh_flag"].GetBool())
+
+        self.solver.SetEchoLevel(self.settings["echo_level"].GetInt())
+        self.solver.Check()
+
+        self.response_function.Initialize()
+
+        self.print_on_rank_zero("::[PotentialFlowAdjointSolver]:: ", "Finished initialization.")
+    def PrepareModelPart(self):
+        super(PotentialFlowAdjointSolver, self).PrepareModelPart()
+       # defines how the primal elements should be replaced with their adjoint counterparts
+        replacement_settings = KratosMultiphysics.Parameters("""
+            {
+                "element_name_table" :
+                {
+                    "IncompressiblePotentialFlowElement2D3N" : "AdjointIncompressiblePotentialFlowElement2D3N"
+                },
+                "condition_name_table" :
+                {
+                    "PotentialWallCondition2D2N"             : "AdjointIncompressiblePotentialWallCondition2D2N"
+                }
+            }
+        """)
+
+        KratosMultiphysics.StructuralMechanicsApplication.ReplaceMultipleElementsAndConditionsProcess(self.main_model_part, replacement_settings).Execute()
```

Comment:
Then you also have to change this:
```suggestion
        ReplaceMultipleElementsAndConditionsProcess(self.main_model_part, replacement_settings).Execute()
```